agent/consul/acl_server.go

```diff
@@ -150,7 +150,12 @@ func (s *Server) ResolveIdentityFromToken(token string) (bool, structs.ACLIdenti
 		return true, aclToken, nil
 	}
 
-	return s.InPrimaryDatacenter() || index > 0, nil, acl.ErrNotFound
+	defaultErr := acl.ErrNotFound
+	canBootstrap, _, _ := s.fsm.State().CanBootstrapACLToken()
```

Comment:
Does this happen on every request where the token is not found? Does that end up being expensive?

Reply:
It is another call to memdb in the case where we are doing a token lookup by secret (i.e. validating the token header used), and failures in resolution, expiration, etc. don't get caught above. But significantly less expensive than than the token lookup itself 8 lines above. It'll get called when using a nonsense token value of the proper format or a token that has been completely deleted. Given that the ACLTokenGetBySecret is already the most expensive piece here, an additional fast lookup to provide clarification in these two scenarios seems worth it.